trieste/acquisition/function/function.py

```diff
@@ -1034,6 +1035,595 @@ def __call__(self, x: TensorType) -> TensorType:
         return tf.reduce_mean(batch_improvement, axis=-1, keepdims=True)  # [..., 1]
 
 
+class BatchExpectedImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
+    """Accurate approximation of the batch expected improvement, using the
+    method of Chvallier and Ginsbourger :cite:`chevalier2013fast`.
+
+    Internally, this uses a highly accurate approximation of the cumulative
+    density function of the multivariate Gaussian, developed by Alan Genz
+    :cite:`genz2016numerical`.
+    """
+
+    def __init__(
+        self,
+        sample_size: int,
+        *,
+        batch_size: int,
+        dtype: tf.DType,
```

Comment:
also, we should be able to work out the dtype from some of the tensors that are given into the function. We dont allow users to specify dtype as an input to any of our other acqs @uri-granta ?

Reply:
Yes, that also sounds good. I've made this change now.